driver.sh

```diff
@@ -103,7 +103,7 @@ mako_reactor() {
   KBUILD_BUILD_TIMESTAMP="Thu Jan  1 00:00:00 UTC 1970" \
   KBUILD_BUILD_USER=driver \
   KBUILD_BUILD_HOST=clangbuiltlinux \
-  make -j"${jobs:-$(nproc)}" CC="${CC}" HOSTCC="${CC}" LD="${LD}" "${@}"
+  make -j"${jobs:-$(nproc)}" CC="${CC}" HOSTCC="${CC}" LD="${LD}" HOSTLD="${LD}" "${@}"
```

Comment:
How in the world did this work? If `LD` is not set, `LD` is defined based on `CROSS_COMPILE`... so we'd be host compiling but cross linking? Seems wrong to me even if it appears to work, I feel like HOSTLD should only be defined if `LD` == `lld`.

Something like this?

```diff
diff --git a/driver.sh b/driver.sh
index 3ca245e..dfdf258 100755
--- a/driver.sh
+++ b/driver.sh
@@ -102,11 +102,12 @@ mako_reactor() {
   KBUILD_BUILD_TIMESTAMP="Thu Jan  1 00:00:00 UTC 1970" \
   KBUILD_BUILD_USER=driver \
   KBUILD_BUILD_HOST=clangbuiltlinux \
-  make -j"${jobs:-$(nproc)}" CC="${CC}" HOSTCC="${CC}" LD="${LD}" "${@}"
+  make -j"${jobs:-$(nproc)}" CC="${CC}" HOSTCC="${CC}" LD="${LD}" HOSTLD="${HOSTLD:-}" "${@}"
 }
 
 build_linux() {
   CC="$(command -v ccache) $(command -v clang-8)"
+  [[ ${LD} =~ lld ]] && HOSTLD=${LD}
 
   if [[ -d ${tree} ]]; then
     cd ${tree}

```

Reply:
Thanks for your suggestion. Indeed it's very strange that the tests did succeed with linker targeting foreign architecture. I changed my commit to use your solution.